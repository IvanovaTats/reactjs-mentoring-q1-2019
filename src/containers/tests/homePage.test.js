import React from 'react';
import { render, cleanup, fireEvent, getByText } from 'react-testing-library';
import 'jest-dom/extend-expect';

import HomePage from '../homePage/homePage';
import SearchResult from '../homePage/searchResult';

const setup = () => {
    const homePage = render(<HomePage />);
    const searchButton = homePage.queryByText('SEARCH');
    return {
        searchButton,
        ...homePage
    }
}

afterEach(cleanup);

test('render a home page', () => {
    const { homePage } = setup();
    expect(homePage).toMatchSnapshot();
});

test('click submit button', () => {
    const { searchButton } = setup();

    fireEvent.click(searchButton, { target: { hasError: true } });

    expect(searchButton).toHaveBeenCalled();
});
