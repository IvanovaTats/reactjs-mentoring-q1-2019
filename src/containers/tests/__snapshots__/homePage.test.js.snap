// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click submit button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="main-menu-form"
      >
        <div
          class="main-menu-input"
        >
          <label
            class="movie-search-label"
            for="movieSearchInput"
          >
            FIND YOUR MOVIE
          </label>
          <input
            class="movie-search-input"
            id="movieSearchInput"
          />
        </div>
        <div
          class="main-menu-buttons"
        >
          <div
            class="search-by-block"
          >
            <span>
              SEARCH BY
            </span>
            <input
              id="titleButton"
              type="button"
              value="TITLE"
            />
            <input
              id="genreButton"
              type="button"
              value="GENRE"
            />
          </div>
          <input
            id="submitButton"
            type="button"
            value="SEARCH"
          />
        </div>
        <div
          class="result-form"
        >
          <div>
            <div
              class="result-form-buttons"
            >
              <span>
                10
                 movies found
              </span>
              <div>
                <span>
                  SORT BY
                </span>
                <input
                  id="releaseDateButton"
                  type="button"
                  value="release date"
                />
                <input
                  id="ratingButton"
                  type="button"
                  value="rating"
                />
              </div>
            </div>
            <div
              class="items-container"
            >
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Guardians of the Galaxy Vol. 3
                  </span>
                  <span>
                    2020-05-01
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Adventure
                    Science Fiction
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Transformers 7
                  </span>
                  <span>
                    2019-06-26
                  </span>
                </div>
                <div>
                  <span>
                    Science Fiction
                    Action
                    Adventure
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/aMETsaNNcDc6g5ZatQtVbySnSaA.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Untitled Avengers
                  </span>
                  <span>
                    2019-04-24
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Adventure
                    Science Fiction
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/yUOJHa9XmB1H0iYodG9Kb3YCc9T.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Shazam!
                  </span>
                  <span>
                    2019-04-05
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Adventure
                    Fantasy
                    Science Fiction
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/6qmsupE0opYPIaBGe7T5D2FBzLs.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    X-Men: Dark Phoenix
                  </span>
                  <span>
                    2019-02-14
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Science Fiction
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/98tbNloMBztAVnWpAznKKVUdi2O.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Journey 3: From the Earth to the Moon
                  </span>
                  <span>
                    2018-12-31
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Adventure
                    Fantasy
                    Science Fiction
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/sshNRCbtU2ON8SiWLycQzr05doX.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Aquaman
                  </span>
                  <span>
                    2018-12-19
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Fantasy
                    Science Fiction
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/A2W75CVz9l8gEeWhQiiftlOCkgi.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Mission: Impossible - Fallout
                  </span>
                  <span>
                    2018-07-26
                  </span>
                </div>
                <div>
                  <span>
                    Adventure
                    Action
                    Thriller
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Jurassic World: Fallen Kingdom
                  </span>
                  <span>
                    2018-06-01
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Adventure
                    Drama
                    Science Fiction
                    Thriller
                  </span>
                </div>
              </div>
              <div
                class="item-poster"
              >
                <img
                  alt="No poster found"
                  class="item-image"
                  src="https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg"
                />
                <div
                  class="item-name_date"
                >
                  <span>
                    Solo: A Star Wars Story
                  </span>
                  <span>
                    2018-05-23
                  </span>
                </div>
                <div>
                  <span>
                    Action
                    Adventure
                    Family
                    Science Fiction
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="main-menu-form"
    >
      <div
        class="main-menu-input"
      >
        <label
          class="movie-search-label"
          for="movieSearchInput"
        >
          FIND YOUR MOVIE
        </label>
        <input
          class="movie-search-input"
          id="movieSearchInput"
        />
      </div>
      <div
        class="main-menu-buttons"
      >
        <div
          class="search-by-block"
        >
          <span>
            SEARCH BY
          </span>
          <input
            id="titleButton"
            type="button"
            value="TITLE"
          />
          <input
            id="genreButton"
            type="button"
            value="GENRE"
          />
        </div>
        <input
          id="submitButton"
          type="button"
          value="SEARCH"
        />
      </div>
      <div
        class="result-form"
      >
        <div>
          <div
            class="result-form-buttons"
          >
            <span>
              10
               movies found
            </span>
            <div>
              <span>
                SORT BY
              </span>
              <input
                id="releaseDateButton"
                type="button"
                value="release date"
              />
              <input
                id="ratingButton"
                type="button"
                value="rating"
              />
            </div>
          </div>
          <div
            class="items-container"
          >
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Guardians of the Galaxy Vol. 3
                </span>
                <span>
                  2020-05-01
                </span>
              </div>
              <div>
                <span>
                  Action
                  Adventure
                  Science Fiction
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Transformers 7
                </span>
                <span>
                  2019-06-26
                </span>
              </div>
              <div>
                <span>
                  Science Fiction
                  Action
                  Adventure
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/aMETsaNNcDc6g5ZatQtVbySnSaA.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Untitled Avengers
                </span>
                <span>
                  2019-04-24
                </span>
              </div>
              <div>
                <span>
                  Action
                  Adventure
                  Science Fiction
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/yUOJHa9XmB1H0iYodG9Kb3YCc9T.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Shazam!
                </span>
                <span>
                  2019-04-05
                </span>
              </div>
              <div>
                <span>
                  Action
                  Adventure
                  Fantasy
                  Science Fiction
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/6qmsupE0opYPIaBGe7T5D2FBzLs.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  X-Men: Dark Phoenix
                </span>
                <span>
                  2019-02-14
                </span>
              </div>
              <div>
                <span>
                  Action
                  Science Fiction
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/98tbNloMBztAVnWpAznKKVUdi2O.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Journey 3: From the Earth to the Moon
                </span>
                <span>
                  2018-12-31
                </span>
              </div>
              <div>
                <span>
                  Action
                  Adventure
                  Fantasy
                  Science Fiction
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/sshNRCbtU2ON8SiWLycQzr05doX.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Aquaman
                </span>
                <span>
                  2018-12-19
                </span>
              </div>
              <div>
                <span>
                  Action
                  Fantasy
                  Science Fiction
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/A2W75CVz9l8gEeWhQiiftlOCkgi.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Mission: Impossible - Fallout
                </span>
                <span>
                  2018-07-26
                </span>
              </div>
              <div>
                <span>
                  Adventure
                  Action
                  Thriller
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Jurassic World: Fallen Kingdom
                </span>
                <span>
                  2018-06-01
                </span>
              </div>
              <div>
                <span>
                  Action
                  Adventure
                  Drama
                  Science Fiction
                  Thriller
                </span>
              </div>
            </div>
            <div
              class="item-poster"
            >
              <img
                alt="No poster found"
                class="item-image"
                src="https://image.tmdb.org/t/p/w500/uJ6OnE3CzGWq6buLINAbdBqa0gV.jpg"
              />
              <div
                class="item-name_date"
              >
                <span>
                  Solo: A Star Wars Story
                </span>
                <span>
                  2018-05-23
                </span>
              </div>
              <div>
                <span>
                  Action
                  Adventure
                  Family
                  Science Fiction
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render a home page 1`] = `undefined`;
